lib/events/eventstest/channel.go

```diff
@@ -44,11 +45,19 @@ func NewChannelEmitter(capacity int) *ChannelEmitter {
 
 func (e *ChannelEmitter) EmitAuditEvent(ctx context.Context, event apievents.AuditEvent) error {
 	e.log.Infof("EmitAuditEvent(%v)", event)
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	case e.events <- event:
-		return nil
+	start := time.Now()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case e.events <- event:
+			return nil
+		case <-time.After(5 * time.Second):
+			e.log.WithFields(logrus.Fields{
+				"event":   event,
+				"elapsed": time.Since(start),
+			}).Info("EmitAuditEvent has been blocked sending to a full ChannelEmitter for a long time")
+		}
```

Comment:
haha, so I decided to test 1000 pings concurrently to check for any issues with heavy concurrent load and I spent half a day debugging absolutely bizarre test failure behavior. Then I realized failures were just because this emitter's channel buffer was filling up and silently blocking RPCs thereafter.